app/models/subscriber.rb

```diff
@@ -0,0 +1,32 @@
+class Subscriber
+  def self.for_discussion(discussion)
+    new(discussion)
+  end
+
+  def initialize(discussion)
+    @discussion = discussion
+    @subscriptions = Subscription.where(discussion_id: discussion.id)
+  end
+
+  def subscribe(users)
+    users.each do |user|
+      create_subscription_for_user(user)
+    end
+  end
+
+  private
+
+  attr_reader :discussion, :subscriptions
+
+  def create_subscription_for_user(user)
+    return if user_already_subscribed?(user)
+
+    Subscription.create(user: user, discussion: discussion)
+  end
+
+  def user_already_subscribed?(user)
+    subscriptions.any? do |subscription|
+      subscription.user_id == user.id
+    end
+  end
```

Comment:
To make the changes suggested in the first comment (returning a list of subscriptions) I'm going to need all the users.

As for the create, are you sure that does not translate to the same amount of queries? I don't think there is exactly a bulk insert in postgres at least. However, we are giving the ORM an opportunity to do as much optimizations as it can, which is good.